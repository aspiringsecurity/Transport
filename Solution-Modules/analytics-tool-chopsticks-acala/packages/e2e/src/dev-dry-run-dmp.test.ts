import { describe, expect, it } from 'vitest'

import networks from './networks'

describe('dev_dryRun dmp', () => {
  it('works', async () => {
    const acala = await networks.acala({
      blockHash: '0x1d9223c88161b512ebaac53c2c7df6dc6bd2731b12273b898f582af929cc5331',
    })
    const params = [
      {
        raw: false,
        dmp: [
          // https://acala.subscan.io/xcm_message/polkadot-2ab22918c567455af3563989d852f307f4cc1250
          {
            sentAt: 14471353,
            msg: '0x02100104000100000b00280b9bba030a13000100000b00280b9bba03010300286bee0d0100040001010070c53d8e216f9c0f2e3b11c53f5f4bf3e078b995d5f0ed590f889f41e20e6531',
          },
        ],
      },
    ]
    const resp = await acala.ws.send('dev_dryRun', params)
    expect(resp.new.system.events).toMatchSnapshot()
  })
})
