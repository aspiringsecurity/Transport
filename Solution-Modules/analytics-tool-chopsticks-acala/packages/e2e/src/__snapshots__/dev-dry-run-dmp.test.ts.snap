// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dev_dryRun dmp > works 1`] = `
[
  {
    "event": {
      "data": {
        "amount": "44,539,728,928",
        "currencyId": {
          "Token": "AUSD",
        },
        "who": "23M5ttkmR6KcnvsNJdmYTpLo9xfc54g8uCk55buDfiJPon69",
      },
      "index": "0x0b0a",
      "method": "Deposited",
      "section": "tokens",
    },
    "phase": "Initialization",
    "topics": [],
  },
  {
    "event": {
      "data": {
        "amount": "2,326,657,466",
        "currencyId": {
          "Token": "AUSD",
        },
        "who": "23M5ttkmR6KcnvsNJdmYTpLo9xfc54g8uCk55buDfiJPon69",
      },
      "index": "0x0b0a",
      "method": "Deposited",
      "section": "tokens",
    },
    "phase": "Initialization",
    "topics": [],
  },
  {
    "event": {
      "data": {
        "amount": "1,151,984,917",
        "currencyId": {
          "Token": "AUSD",
        },
        "who": "23M5ttkmR6KcnvsNJdmYTpLo9xfc54g8uCk55buDfiJPon69",
      },
      "index": "0x0b0a",
      "method": "Deposited",
      "section": "tokens",
    },
    "phase": "Initialization",
    "topics": [],
  },
  {
    "event": {
      "data": {
        "amount": "6,761,612,127",
        "currencyId": {
          "Token": "AUSD",
        },
        "who": "23M5ttkmR6KcnvsNJdmYTpLo9xfc54g8uCk55buDfiJPon69",
      },
      "index": "0x0b0a",
      "method": "Deposited",
      "section": "tokens",
    },
    "phase": "Initialization",
    "topics": [],
  },
  {
    "event": {
      "data": {
        "dispatchInfo": {
          "class": "Mandatory",
          "paysFee": "Yes",
          "weight": {
            "proofSize": "0",
            "refTime": "260,558,000",
          },
        },
      },
      "index": "0x0000",
      "method": "ExtrinsicSuccess",
      "section": "system",
    },
    "phase": {
      "ApplyExtrinsic": "0",
    },
    "topics": [],
  },
  {
    "event": {
      "data": {
        "count": "1",
      },
      "index": "0x1e04",
      "method": "DownwardMessagesReceived",
      "section": "parachainSystem",
    },
    "phase": {
      "ApplyExtrinsic": "1",
    },
    "topics": [],
  },
  {
    "event": {
      "data": {
        "amount": "4,099,998,328,990",
        "currencyId": {
          "Token": "DOT",
        },
        "who": "23RTYMXEKv7MkSh6CPfyFVYXweCWUF6WncMY1jzhxaw2pV7b",
      },
      "index": "0x0b0a",
      "method": "Deposited",
      "section": "tokens",
    },
    "phase": {
      "ApplyExtrinsic": "1",
    },
    "topics": [],
  },
  {
    "event": {
      "data": {
        "amount": "1,671,010",
        "currencyId": {
          "Token": "DOT",
        },
        "who": "23M5ttkmR6KcoTAAE6gcmibnKFtVaTP5yxnY8HF1BmrJ2A1i",
      },
      "index": "0x0b0a",
      "method": "Deposited",
      "section": "tokens",
    },
    "phase": {
      "ApplyExtrinsic": "1",
    },
    "topics": [],
  },
  {
    "event": {
      "data": {
        "messageId": "0xca342f7f7a0756371bce233b208eef2c5d5311c90615f896ad38254f3c8acf82",
        "outcome": {
          "Complete": "800,000,000",
        },
      },
      "index": "0x3502",
      "method": "ExecutedDownward",
      "section": "dmpQueue",
    },
    "phase": {
      "ApplyExtrinsic": "1",
    },
    "topics": [],
  },
  {
    "event": {
      "data": {
        "dmqHead": "0x6db9e5ea6d489bbd516e7923c0534a86c4e61207e523cfd1eb4f7ed3c2c350f7",
        "weightUsed": {
          "proofSize": "0",
          "refTime": "800,000,000",
        },
      },
      "index": "0x1e05",
      "method": "DownwardMessagesProcessed",
      "section": "parachainSystem",
    },
    "phase": {
      "ApplyExtrinsic": "1",
    },
    "topics": [],
  },
  {
    "event": {
      "data": {
        "dispatchInfo": {
          "class": "Mandatory",
          "paysFee": "No",
          "weight": {
            "proofSize": "0",
            "refTime": "98,974,000",
          },
        },
      },
      "index": "0x0000",
      "method": "ExtrinsicSuccess",
      "section": "system",
    },
    "phase": {
      "ApplyExtrinsic": "1",
    },
    "topics": [],
  },
]
`;
